src/mon/MgrMonitor.cc

```diff
@@ -580,8 +580,8 @@ void MgrMonitor::tick()
   if (!pending_map.available &&
       !ever_had_active_mgr &&
       should_warn_about_mgr_down() != HEALTH_OK) {
-    dout(10) << " exceeded mon_mgr_mkfs_grace "
-             << g_conf->get_val<int64_t>("mon_mgr_mkfs_grace")
+    auto mon_mgr_mkfs_grace = g_conf->get_val<int64_t>("mon_mgr_mkfs_grace");
+    dout(10) << " exceeded mon_mgr_mkfs_grace " << mon_mgr_mkfs_grace
```

Comment:
does this make any difference?

Reply:
Using get_val<> wouldn't cheap, isn't it?

Reply:
and

- in the worst case, if debug-mon>=10, your change does not improve the performance. as `get_val<>` always gets called
- in the best case, if debug-mon<10, your change hurts the performance, the return value of `get_val<>()` is dropped on the floor, and is not used.

Reply:
i got your point now. i didn't properly understand the context of your comment. sorry for that...